feat(data): Afterburner/Accessory slot system

Zitchas opened the pull request:
**Balance**

## Summary
By request, this is strictly the afterburner section of #9963 , the engine slot PR. People have pointed out to me that while there may be some challenges with the overall engine slot system, they like and support the idea of the afterburner slot system as described therein. So, here it is.

Note: I have renamed it the "thruster accessory slot"  to reflect the fact that while right now only afterburners use it, I hope that there will be other outfits in the future that take up all or part of these slots to modify the thruster they are attached to, just like the afterburner does. 

Note2: This does not touch code. 

## Details
- Ever thruster provides 1 "thruster accessory slot"
- Every afterburner consumes 1 "thruster accessory slot" 

## Usage examples
`"thruster accessory slot" 1`

Negative is using up a slot, positive is adding a slot. Can be decimals, but this PR only utilizes integers. Smaller outfits that affect thrusters could be smaller.

## Testing Done
Played the game with it.

## Save File
This save file can be used to test these changes:
Any pilot, any ship, anytime, anywhere. Doesn't change any ship stats, so won't interact problematically with ships. Other than ships that start off with more afterburners than they have thrusters. Which exists, and will be resolved shortly.

Hurleveur reviewed (approved):
(Since there is no code this is only my personal opinion)
I really like the idea of this, I dont feel like it makes sense to have an AB without a thruster, we could push it further with bigger AB taking more slots and bigger engines giving more, but that could be left for later.
This makes it harder to spam AB without consequences, making the cost a little bit higher, which is a good thing, but doesnt make it too restrictive either. There could even be a special AB that doesnt require a thruster

We should probably update the desc of ABs to include that they need to be attached to an engine, and if ppl still get confused a later PR could make this outfit attribute be shown

Zitchas commented on data/_ui/help.txt:
```diff
@@ -141,7 +141,7 @@ help "navigation"
 help "outfitter"
 	`Here, you can buy new equipment for your ship. Your ship has a limited amount of "outfit space," and most outfits use up some of that space.`
-	`Some types of outfits have other requirements as well. For example, only some of your outfit space can be used for engines or weapons; this is your ship's "engine capacity" and "weapon capacity." Guns and missile launchers also require a free "gun port," and turrets require a free "turret mount." Also, missiles can only be bought if you have the right launcher installed.`
+	`Some types of outfits have other requirements as well. For example, only some of your outfit space can be used for engines or weapons; this is your ship's "engine capacity" and "weapon capacity." Guns and missile launchers also require a free "gun port," and turrets require a free "turret mount." Also, missiles can only be bought if you have the right launcher installed. Afterburners and other  ...
```
@endless-sky/writing-patrol 

This line is the only user-visible text change. I'm not set on the wording used here, I just feel that there should be an addition regarding it.

Alternative:
"Afterburners and other thruster modifications must be installed after the thruster they are attached to."

Quantumshark replied:
One option would be to write some kind of "afterburners require an installed thruster" boilerplate text and stick it on all afterburners, like the "this ship requires a bay and lacks a hyperdrive etc" boilerplate on fighters and drones or the "this is ammunition, you can't install it unless you have something to fire it from" boilerplate on missiles.

Something along the lines of
`Afterburners cannot be installed on their own. They require a thruster to be installed first.`, but possibly slightly more eloquent.

Zitchas replied:
Yes, as per Hurleveur's comment, already planning on adding such a line. Do you think that would be sufficient?

Quantumshark replied:
I think that would be sufficient to communicate the idea to players, yes. If it isn't, then we aren't communicating missile capacity or the concept of a fighter/drone well enough to the players currently either.

xX-Dillinger-Xx replied:
> Yes, as per Hurleveur's comment, already planning on adding such a line. Do you think that would be sufficient?

As a player I think it would be sufficient.

EjoThims replied:
Agreeing that the boiler plate is best way to go, but that doesn't mean we shouldn't also include a reference in the tutorial teaching about it.

Zitchas replied:
Alright. I have added a boilerplate to all the afterburners except for the asteroid, which received an addendum noting that unlike regular afterburners, it *does* have the capacity to be installed by itself, as described in its description.

Zitchas commented:
Good point. I will add a line to the afterburner descriptions.

Quantumshark reviewed (commented):
I'm not necessarily opposed to this change.

I do think though, that if afterburner-only builds are too strong, the approach we should be taking is to look at what the issue with afterburners' stats are that's making them overperform in a niche in which they aren't intended to, rather than simply imposing an artificial limit on them.

That aside, I'm not entirely sure it makes sense to simply have all afterburners take one slot and all engines give one slot. Does it make sense that you can mount a stellar-class afterburner on the end of an X1700 ion engine? Does it actually solve the problem of afterburner-only builds if you can circumvent the restriction just by adding a small thrust module or another similarly small thruster?

Maybe it'd be better to have some kind of `afterburner capacity` provided by thrusters, based on their size, and have afterburners consume an amount of `afterburner capacity` based on their size. That'd also provide a new dimension in which to differentiate both thrusters and afterburners - a thruster set could be relatively poor, but a better base to install afterburners on, or the reverse, or a particularly good afterburner could require more `afterburner capacity` than its size would suggest.

Zitchas commented:
I'm intrigue regarding such complexity suggested by having afterburner capacity, but I was under the impression that even the most simplistic 1 thruster, 1 afterburner lack of limitations was a long shot; never mind realitistic limitations based on size. I'm also dubious about the idea that 1 large thruster can stack, say, 4 small afterburners on it.

An alternative option would be to ditch the current afterburner functionality entirely, and instead have it be strictly multiplicative. Afterburner A gives "current thrust" x 0.25 additional thrust, for instance. Install it on a big thruster, it gives bigger amount. Not sure if that's a good thing, but it is an option. 

For that matter, there's also the fact that chances are pretty good that an effective afterburner design for a plasma thruster is probably useless on an ionic or atomic thruster; and vice versa.

As far as that goes, yes. I do think that imposing at least a minor cost on the pure-afterburner builds of requiring them to sacrifice a bit of engine space to get the requisite number of X1700 or small thrust module is a clear net benefit. It might not solve it entirely, but it tones down the extremes that can be reached by it. That's a positive for the game balance. In the same vein, I don't particularly want to completely eliminate things like this. Figuring out a clever high-performance with a bit of risk build is a generally a good thing, I just think that it would be beneficial to cap off the utmost extreme ends of it. 

As for the root problem:

The entire purpose of afterburners is to enable expenditure of resources to enable an extremely high performance/space compared to standard thrusters. 

Most normal thrusters do not consume fuel at all, only energy and produce heat with moderate space utilization.

Afterburners consume fuel and energy, and produce heat, with a low space utilization.

Fuel is generally considered the main limiter of afterburners, and everyone regenerates that for free, it's just a question of how fast.

Energy and Heat are the secondary limiters, and likewise everyone regenerates *those*, and they are pretty close to free (just some space costs) too. 

Looking at thrusters, they mostly all require energy and generate heat, too. And I know there's been talk of having a fuel-using thruster as well.

Some people look at afterburners and consider all consumption aspects to be manageable regardless of magnitude, and only the actual thrust/ton matters. Others take it all into account in varying degrees. So to some extent, balancing is problematic because most of the goalposts aren't lined up.

So at at first glance, afterburners vs thrusters is problematic because one is comparing "mostly constantly working thruster" vs "somewhat less constantly working afterburner." The only real difference is uptime: Which one can be run the longest before running out of resources? If one can get that to 100% for both, then the difference comes down to which one gives more thrust/space. 

If both can be installed infinite numbers of times without relation to each other, then they are both effectively just different kinds of thrusters.

ziproot reviewed (changes requested):
1. The Mammoth is still broken here
2. There should probably be something telling you that you don’t have enough engines installed to install the afterburner

mOctave commented:
I can't say I like locking afterburners behind engines. As someone who doesn't max my builds out, fuel tends to be a real concern for my afterburner-only ships. The only time I ever use afterburners is when I'm replacing an engine with them, because otherwise they take up too much space to be worthwhile since combat still tends to be a slugging match. An afterburner-only ship requires much more piloting skill than a thruster ship: you can't just trust autopilot to land for you, since it'll be really inefficient in doing so and might cause you to overshoot. Overall, I've never had the problem of afterburners being OP that more dedicated players seem to have. For me, installing an afterburner is a solution to give you more space where the trade off is having to pay more attention to how you fly your ship.

If you remove afterburner-only builds, I can't see myself ever using afterburners, because I don't tend to max out ramscoop, and the afterburner doesn't accomplish anything that a thruster could instead. Sure, afterburners locked behind thrusters might be more realistic, but I think that would detract from gameplay.

I'd be interested to see, actually, what percentage of the player base uses thrusters vs afterburners vs both on their builds.

Hurleveur commented:
> the afterburner doesn't accomplish anything that a thruster could instead.

The intention of the AB is to be combined with a thruster and to give a boost in emergencies. You could make a case of the ionic AB being closer to an engine
I can get interceptor speed on world ships with AB only, I think it's an issue with both things though, but fuel isnt hard to get when you realise you dont need to spam the AB, you just get the speed you need in a few secs then stop thrusting until you have to (like you can do with all engines to a lesser degree)

EjoThims commented:
I do very much agree that nerfing ABs into uselessness is not the proper direction; another way to limit them is the most appropriate means of keeping them actually viable at all. I also disagree that chopping large ABs entirely is a good move, as when -used as intended-, they do still deliver the intended result.

At the same time, I also definitely think this should lean more into how @Quantumshark outlines it here, because, as proposed, this is a huge relative buff to small engines (letting you stack up AB slots) -and- really only a "smallest possible thruster  tax" on AB only builds.

I would go one step further in this toned down middle ground alternative, though, and also include (preferably fewer by ratio) slots on turning as well as thrusters*. This would re-allow for thrusterless builds, yes, but in a much more limited capacity than today. Especially in larger ships, there would still actually be optimization choices between -enough- turn for the ABs you want and still using at least some smaller thrusters as well, while also being more figuring to speedy small builds using just turning and AB combos (which IMO shouldn't be strictly impossible when the primary problem is still large ship abuse of that combo).

xX-Dillinger-Xx commented:
My vision of an afterburner:
-  can not be used as a standalone engine.
- must be attached to an engine of similar design and should mirror the parent engine group ( output vs. size ratio )
  - ionic afterburner on ionic engines,  remnant on remnant,  etc. etc.
    - there could be some mix and match allowed. for example, you could but a Korath Asteroid Class on a human Plasma Thruster they both use Plasma technology.
  - only one afterburner per engine
- must provide a boost to the base engines thrust.
  - the amount of boost has to be proportional to the engine it is attached to. A _percentage_ of base engine thrust makes the most sense. 
- must be short use only, for emergencies. 
  - comes at a significant cost in the form of high fuel consumption, high energy consumption, and generate excessive heat. It should also be a varying degree of cost based on the design and also increase with the total amount boost thrust added.  
- uses engine space and outfit space. Varies based on design, not on perceived output.  
  - examples: Korath Asteroid Class contains a ramscoop and a fuel pod and should require more space then an ionic.

EjoThims commented:
*as an aside, this would also make more sense to do now so we can later include things like "turning boost" (or other similar alternative uses if these engine slots), which I'd love to see as an alternate competitor with reverse thrust for "back" functionality - let it kick in specifically with that key press for an extra tight turn around that could be utilized for kiting, jousting, and even landing in a hurry.

mOctave commented:
Thinking about what warp said in closing #9963, is there even an afterburner problem? Or do we just need to do a rebalance of the Korath and Remnant afterburners. Bellows definitely seems OP. I think the human afterburners tend to be pretty balanced. What if we increased the Bellows to 30 or so but actually decreased its heat and energy requirements? Then it would still fit with the Remnant's higher tier without being so easily offset. Not sure how to balance out the P/C class Korath afterburners, but I'm sure it could be done.

ziproot commented:
Bellows will be balanced in #9693

Zitchas commented:
> While this might make more logical sense, I'm not sure it solves the whole "running with only afterburners" issue. A Small Thrust Module is what, 9 tons? That's basically negligible when you're getting >50% outfit space gains by replacing your main thrusters with afterburners.

This moves afterburner-only from "completely non-sensical abuse & OP" to merely "eyebrow-raising combination". And by that, I mean it drops the total performance from the realms of "Feels like I figured out how to break the game" to "I figured out a really powerful trick."

As an illustration, I just took a stock Behemoth, downgraded its steering to X2200, sold its thruster, and filled its engine space with Bellows. This results in the max speed of 6410 and an acceleration of 2962-4373.

Now, if I do the exact same thing, but I have to maintain a 1:1 ratio of small thrust modules to bellows, Which works out to 4 small thrust modules and 4 Bellows. I had to use Delta to see in-flight speeds to get the numbers, so probably not 100% accurate to vanilla as the physics is a bit different, but in any case, the final top speed I achieved was ~2734. And that took almost 2000 fuel, and a decent chunk of time, too.

I don't know about you, but a 6410 down to 2800 for an afterburner abused Behemoth is *much* more reasonable. More than fast enough to make *players* happy with the loophole they found; but still limited to a much more reasonable level. And that's the extreme case for players that have access to two different alien outfitters. Replace the "small thruster module" with an X1700, which is bigger; and the Bellows with any of the other afterburners, all of which are lower thrust; and that number will shrink quite a bit more. Sounds pretty effective to me.

ziproot commented:
2800 is still very fast, especially for a slow ship like the behemoth.

The shield beetle can go at >5000 max speed and 1800 acceleration using the same turning values of around 15 (X2200 ion steering) and with 6 bellows + 6 small thrust modules. This is assuming the bellows balance changes are made, otherwise it would be almost 6000 max speed and 2000 acceleration. The reason why this is balanced is because that would use 468 fuel in one second, which is almost all of the shield beetle’s fuel, at the cost of turning.

Without slots but with the balance changes, you could fit 11 bellows afterburners, which has 9300 max speed and 3000 acceleration, at the cost of turning and using all of the shield beetle’s fuel in 0.58 seconds. That would basically be only good for sending your ship flying into space with no way of stopping. With the behemoth, you could stop thanks to the reverse thrusters, but you would still have to navigate by reverse thrust to land. 

Basically, afterburners are already naturally balanced by their fuel usage, and don’t need an artificial limitation. There are a few, like the bellows and korath afterburners, that are unbalanced in master, but the bellows is going to be looked at.

The korath afterburners come in size categories, so you could just fit a small thrust module and a planetary class afterburner rather than just the planetary class afterburner by itself, or for the shield beetle, a stellar class afterburner and a small thrust module for 3000 max speed, rather than just a stellar class afterburner.

So this only really works for “balancing” small afterburners, and only really the bellows in particular, as human afterburners aren’t too much of a problem, at which point you might as well just balance the bellows, which is what is being done.

Saugia commented:
Something I feel is warranted in discussing is that there has been a desire to focus more on what's currently within the game and to work with that before adding more mechanics or facets to balance the game around. Balancing modules in a game compared to one another is not a bad thing as everyone is making it out to be - this isn't an overarching comparison of fun vs. balance.

One of the most important aspects of Endless Sky is that rewards or outfits aren't the focus of the game, the story is. So outfits shouldn't be balanced around the idea that they're special or one-ofs. This isn't to say they can't be somewhat stronger compared to other more common outfits, but I feel like a lot of this discussion around the bellows, other afterburners, balancing in general, etc., has been exaggerated more than it needs to be.

Unless we're gutting an afterburner or stats in half or more, in which most changes relevant to discussions like this are only slight adjustments, there shouldn't be a problem. The outfits will still have their relative comparisons. I don't think overall restricting how the engine system works is going to fix the perceived problem. I don't have a strong opinion on this quite yet, but I don't think this is a stand-in versus adjusting afterburners or outfits in general, but another way to go about outfitting overall.

Saugia commented:
So I suppose I'll ask what the intention is then. From what I've gathered, it could be one of three (or all) things:

Is it limiting afterburner customization to give smaller ships a purpose?
Is it avoiding builds that include several afterburners stacked?
Is it to avoid a need to balance afterburners individually?

If it's the first one, I think that the purpose of small ships is subjective. Currently, we have a mass PR pending review that we're interested in, that makes small ships faster and larger ships slower, giving more weight to their speed differential. We may have the Shield PR too with higher regen on smaller, slower on larger ships (which I also think is good). So, the general idea seems to be that small ships have a purpose with their speed, which is what these changes are working on fixing.

If it's the second one, I think this is moreso in terms of what specifically the player does. No AI ship (excuse me Tehhowch) has a build that unnecessarily stacks afterburners or uses them in a way that would make a large ship ridiculously fast. Now, it can be argued that some afterburners are notably strong, and can still be tuned down, and still be strong, if that's their intent. I will admit, however, I'm somewhat indifferent here. If you do want to use the EV comparison, the few engine tuning outfits in EV would never put a large ship past the speed of a smaller ship, yes. So I understand where you're coming from there. My question is, is this a major concern with player builds in ES? I feel like these types of builds are rather rarely used, and in most cases are more as something silly rather than an intended advantage.

If it's the third one, again, I feel like that we can still adjust very strong afterburners and they will still be strong. The Bellows + Korath afterburners are good afterburners, they're meant to be strong. However, if their numbers are very far away from that of others in the game and in comparisons, they can still be adjusted. If the adjustments don't change their comparative strength to other burners in the chart, then they still hold the same value compared to them. This is separate than whether or not afterburners define a small ship's purpose.

xX-Dillinger-Xx commented:
The really issue and what I believe to the inspiration behind this PR is players are swapping engines/thrusters for afterburners. The why is simple, big ships are too slow, but that's a different topic. 

As I already stated ABs are not and should not be stand alone engines. They are a supplement to an engine designed for emergency use, plain and simple, and therefore their output should in direct relation to the base engines output. The simplest solution is a percentage based one.  Has anyone tried using a percentage yet to see what it looks like? Come on guys, now you can fire up those spread sheets. 😏 

Will this PR cause issues for those that want AB only builds? Yes, most definitely, but it will also allow for future growth in the form of other currently un-thought of uses, and to me, the Pros far out weigh the Cons. If the current implementation of ABs is crap, get rid of it and build something better. You can polish a turd all you want it's still a turd.

Saugia commented:
> The really issue and what I believe to the inspiration behind this PR is players are swapping engines/thrusters for afterburners. The why is simple, big ships are too slow, but that's a different topic.

It's worth nothing that afterburners are built with the intent that that they can be used instead of engines. We've explicitly [added code and mechanics to the game in the past](https://github.com/endless-sky/endless-sky/pull/3063) so this is feasible.

I still haven't seen very many cases where players go out of their way to stack only afterburners for efficiency to the point where its breaking engine balance. Players avoid builds like this because afterburners, like other outfits, use a finite resource. At some point, stacking afterburners will require space for fuel or fuel generation, which is already making your ship less efficient than if you were to have thrusters.

I don't find it appropriate to refer this to a "turd" (and that's not really an appropriate comparison to begin with). This isn't a comparison of something bad vs good, it's a discussion on an outfit's intention and ships' relativeness to it.

xX-Dillinger-Xx commented:
>If the current implementation of ABs is crap, get rid of it and build something better. You can polish a turd all you want it's still a turd.

Fine, I'll put it differently. Sorry, it was meant as a humorous analogy.
If the current implementation of ABs is ~~crap~~ problematic, get rid of it and build something better. It's current state will only take you so far, and will always have the same problems unless it's replaced.

mOctave commented:
Also, Oxford's definition of "afterburner" is as follows:

> an auxiliary burner fitted to the exhaust system of a turbojet engine to increase thrust.

It's not like we're even using turbojets, and the engines in-game are way more OP than anything you could expect from the technology they supposedly use. So why do we care about matching the dictionary definition of afterburner as closely as possible?

LorenzoBolla commented:
> I wonder if we might be better off with, instead of slots-per-engine for afterburners, giving them some sort of afterburner capacity equal to the amount of engine space they take up, and having afterburners use that instead of engine space (would probably want to shrink them all as well). That might mitigate the engine space bottleneck all ships currently face, but I'm not sure it's the most elegant solution
> 
> As for the currently single (as far as I know) instance of a ship variant relying exclusively on afterburners, seeing as that's a fairly recent and isolated addition, I wonder if it might just be better to treat the Asteroid class AB as a fuel consuming engine, the way all afterburners are currently set up as.

Azure, that's a **very** elegant solution. 

Keeps current features (fuel-using engine), makes a (i believe almost entirely) "ungameable" limit to afterburner only builds, and as a plus, mitigates that bottleneck, making ABs use only the less valuable general outfitting space.

Neat!

Zitchas commented:
A) If an afterburner (such as the Korath asteroid one) are designed and used as being a thruster-that-consumes-fuel replacement for thrusters, then it should be reclassified as a thruster that consumes fuel. It is, in contrast to our existing thruster set, more of a rocket and less a motor, but that's fine. Diversity is good. There's nothing that says that all fuel-consuming-thrusters need to be classified as afterburners, after all. We could even leave it *called* an afterburner, and just adjust its description to note that it is called an afterburner due to its similarities and fuel consumption, but it is a fully self-contained thruster and thus is treated like one instead of like an afterburner. It introduces a bit of confusion, but it retains full compatibility; and doesn't break anything.

B) Swapping from a slot system to a "accessory space" that the afterburner takes up a varying amount of depending on its size sounds quite interesting. It is more complex, but does allow for people to opt for using a smaller-than-maximum-size afterburner in conjunction with other sorts of thruster modifications if they so choose. Greater flexibility in that regards is good. 

c) Not so fond of the idea of allowing people to get a big thruster and then stack 4 afterburners on it, though. That brings back the problem we have with thrusters themselves, that bigger thrusters have to be better than small ones, because if this wasn't the case, everyone would just stack up small ones. That could be alleviated, however, by careful value assignation. For instance, have thrusters provide an accessory slot ranging from 0.76 up to 1.5 depending on size, and then give afterburners a value that range from 0.76 at the small end to 1.5 at the huge end. That ensures that they can't be stacked, but does allow smaller afterburners to be used as a tradeoff if the player wants that extra slot space to use for as-yet uncreated outfits.

D) Removing afterburner's engine space requirement seems quite appropriate if the afterburners are explicitly attaching to a thruster instead of being a stand alone outfit. I could even see removing the outfit space requirement; or at least significantly reducing it. Probably shouldn't be removing outfit space entirely unless there was more of a thruster-accessory-ecosystem, though, otherwise it becomes a "Well, of course we have the biggest afterburner our thruster can handle? What else would we do with that space?" Of course, I would very much like to see such an ecosystem, and creating space for it could encourage it to develop.

Zitchas commented:
OK, as per earlier suggestions, I have added a boilerplate to most of the afterburners:
> "Afterburners do not contain all the components required to be stand-alone propulsion, and thus must be installed on a thruster."

With the exception of the Korath Asteroid-class afterburner which specifically calls out being designed to be used stand-alone in its description, so it received the following addition in place of the above boilerplate:

> "Despite being labelled an afterburner and connected to that control system on ships, this outfit does not fill the afterburner slot on a thruster."

I have also removed the slot requirement on the asteroid-class afterburner.

Zitchas commented:
Unless there's a strong interest in actually renaming the asteroid class afterburner to reflect the fact that it is a compact fuel-burning thruster instead of an afterburner right now, I'd say that is something that can be left for a stand-alone PR. 

As for the state of things:

Right now, this purely introduces an afterburner slot that is provided at a rate of 1 per thruster, and consumed at a rate of 1 per afterburner, with the asteroid-class being the exception.

Azure has expressed an interest in ditching the engine space and slot requirements of afterburners in favor of having thrusters give a variable amount of afterburner space, which is consumed by the afterburners. This would restrict people from installing giant afterburners on tiny thrusters. There's a couple ways that this could be done.

I personally like having this refinement, but I do not know the general sentiment is such that I should implement it, or not. I'm inclined to do so, as I both like it and it provides a mechanism for allowing people to deliberately pick smaller afterburners so they can use that extra accessory space to fit other kinds of thruster modifications. (while ensuring that the amounts are calibrated to prevent stacking multiple afterburners on one thruster)

Quantumshark commented:
> For instance, have thrusters provide an accessory slot ranging from 0.76 up to 1.5 depending on size, and then give afterburners a value that range from 0.76 at the small end to 1.5 at the huge end. That ensures that they can't be stacked

Would the ability to use a handful of small afterburners rather than just one on a large thruster actually be problematic? If you're using most of your engine space on a big thruster, you're already fairly heavily limited in terms of how much space you've got left for afterburners. I don't think the ability to install, say, two volcano-class afterburners rather than one with an A520 thruster would really be an issue.

Zitchas commented:
We've already got the "bigger versions must be better in most if not all ways because otherwise why bother having them at all" problem with thrusters, we don't need to *enable* continuing to have that problem with afterburners. If the 1:1 ratio of afterburners to thrusters is maintained, that enables a wider variety of potential outfits. Likewise, once this mod space exists, people can create other sorts of things to use it. 1 thruster + 0 or 1 afterburner + mods that affect one or both of those is quite a bit more interesting than 1 thruster + 1 or more AB.

xX-Dillinger-Xx commented:
Like I already said, being able to add more then 1 afterburner per engine doesn't make sense for real life technical reasons. This effects the immersion level of players that know that. I know how much people hate it when people like me bring up realism, but the more real, the more believable and that allows the player to stay immersed in the ES world. What breaks that is when well known real life technologies are use or described incorrectly.  For me, and I'm sure others, my thoughts immediately go to "well that doesn't make any sense", and now I spend more time thinking about that then the game itself. Obviously we can't make everything hyper realistic or then it becomes a sim and not a game, but we should try our best to keep it as believable or explainable as possible without going to far in the wrong direction. 

If you want them to be additional rocket thrusters that can be stacked then change the name. "Rocket Thrusters" is a better fit for what we are _currently_ calling afterburners.

Bubadark commented:
last I tested it escorts do not use afterburners unless in an afterburner only build. so if my lest test is still true this change would eliminate the ability to use afterburners on escorts.
I feel having a duty cycle so afterburners especially the bellows can only fire for say 15 frames then have a 150 or 450 frame cooldown would be cool or having afterburners apply a multiplier to the engines that allready exist like in the EV games. but all of these options require coding skills that i dont posses.

ziproot reviewed (changes requested):
There are still some ships that will have negative slot space if this PR is merged, for instance, the ship you fight in the free worlds epilogue to unlock the plasma repeater is an afterburner only ship, along with the already mentioned olofez. As far as I can tell, those are the only two ships that will have negative slot space.

EDIT: Also, I think the accessory slots don’t have tooltips.

TheGiraffe3 reviewed (approved):
This will work a lot better. Right now, for example, a Kimek Spire with three Caldera Afterburners goes crazy fast-especially if that ship was bought pre-0.10.7…

Quantumshark commented:
I think the real problem here is that afterburners' stats encourage them to be used in a way that doesn't fit with the 'intended' use case.
I don't think having the _option_ to use a build with only afterburners is a problem, I think the fact that certain afterburners are powerful enough that such builds are vastly better than other builds using regular engines is the problem.
This PR doesn't really fix that problem, it just adds a new restriction which can be circumvented by installing one tiny thruster in addition to your afterburner.
Really, the solution is to either nerf the afterburners that are too strong, or implement some new mechanic like #10152 that further differentiates afterburners from thrusters.

TheGiraffe3 commented:
> I think the real problem here, if anything, is that afterburners' stats encourage them to be used in a way that doesn't fit with the 'intended' use case.
> I don't think having the _option_ to use a build with only afterburners is a problem, I think the fact that certain afterburners are powerful enough that such builds are vastly better than other builds using regular engines is the problem.
> This PR doesn't really fix that problem, it just adds a new restriction which can be circumvented by installing one tiny thruster in addition to your afterburner.
> Really, the solution is to either nerf the afterburners that are too strong, or implement some new mechanic like #10152 that further differentiates afterburners from thrusters.

That is all completely true. However, this does help a bit. If the fuel required for using afterburners is, say, doubled, that would probably also help a lot.

Quantumshark commented:
As I've said, I don't think this PR is a good way to solve the problem it sets out to solve. The same balance problems that exist around afterburner-only setups still exist with this PR, they just shift to afterburner + tiny thruster setups. If we have decided that we want to change the way people use afterburners, we should do so by giving afterburners stats that make people _want_ to use them in the way we think they should be used, not by simply barring players from using them in other ways.

Beyond that, it's a general principle of ES that on the whole, we prefer to limit outfits by size rather than by count. There are exceptions - weapons being limited by hardpoint, for example, but that's mainly about differentiating ships from each other by giving them different, often very different, hardpoint counts. This change doesn't provide more differentiation of ships or outfits, since it's universally one slot per engine, so you're restricting player options without really providing a benefit.

With regards to the question of whether it makes sense in-universe for afterburners to be able to exist without an attached engine:
Afterburners in the sense that they exist on earth wouldn't work in space, so it's evident that 'afterburner' here is just a name, not actually a description of how they function. I think it makes less sense to say that afterburners attach to and require engines, and that the same afterburner can work identically whether it's attached to a tiny ion engine or a colossal plasma engine, than to just say that afterburners are independent engines distinguished primarily by the way they're used rather than necessarily by their mechanism of action.

With all these things considered, I don't think the changes in this PR would be an improvement to Endless Sky.
